src/utils.jl

```diff
@@ -72,16 +72,10 @@ end
 
 allzeros(x::ZeroVector) = true
 
-function allnonneg(x::Array{T}) where T<:Real
-    for i = 1 : length(x)
-        if !(x[i] >= zero(T))
-            return false
-        end
-    end
-    return true
-end
+allnonneg(xs::AbstractArray{T}) where T<:Real = all(x -> x >= zero(T), xs)
```

Comment:
```suggestion
allnonneg(xs::AbstractArray{<:Real}) = all(x -> x >= 0, xs)
```
Comparisons with 0 work across types